game/src/components/health.rs

```diff
@@ -0,0 +1,31 @@
+use bevy::prelude::*;
+
+///pointer struct to mark entity as alive
+#[derive(Component, Debug)]
+pub struct Alive;
+
+///pointer struct to mark entity as dead
+#[derive(Component, Debug)]
+pub struct Dead;
+
+///Health component to track entity health
+#[derive(Component, Debug)]
+pub struct Health {
+    pub maximum_health: u32,
+    pub current_health: u32,
+}
+
+impl Health {
+    pub fn new(maximum_health: u32, current_health: u32) -> Self {
+        Self {
+            maximum_health,
+            current_health,
+        }
+    }
```

Comment:
Now that I think about it, instances where we would have a different starting health than the maximum should be pretty few and far between. Change this method to only take the maximum_health as an argument. 
## Optional
If you wanna be fancy, you could add methods to modify the `current` field on instantiation like so:
```Rust
fn with_current_health(mut self, health: u32) -> Self {
    self.current = health;
    self
}
```
You can also add checks to prevent having current hp be higher than the max.
This way you can call it like so:
```Rust
let my_hp_1 = Health::new(100);
let my_hp_2 = Health::new(100).with_current_health(50);
```
I think this pattern is pretty useful for simple data types. A bit like the builder pattern.

Reply:
Done